src/Microsoft.AspNetCore.Server.IISIntegration/Server/IISHttpRequestBody.cs

```diff
@@ -39,7 +39,18 @@ public override int Read(byte[] buffer, int offset, int count)
 
         public override unsafe Task<int> ReadAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
         {
-            return _httpContext.ReadAsync(buffer, offset, count, cancellationToken);
+            if (buffer == null)
+            {
+                throw new ArgumentNullException(nameof(buffer));
+            }
+            if (count == 0)
```

Comment:
This shouldn't throw.

Reply:
Stream.Read/Async should not return 0 until the end of the body. You need a non-zero buffer to do that.